### PR TITLE
feat(person): add spouse to person

### PR DESCRIPTION
**Pull request by jnizet**

### CODE REVIEW

File: backend/src/main/java/org/globe42/domain/Couple.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.SequenceGenerator;
+
+/**
+ * A couple (typically, marriage or PACS) between two persons.
+ * This is not designed as a OneToOne association between persons, because the association is optional and needs
+ * to be bidirectional. This would either require byte-code enhancements or additional queries to know if the
+ * association exists, and this is cumbersome or inefficient because we almost always don't care about this association.
+ *
```
Comment (acrepet) [inline backend/src/main/java/org/globe42/domain/Couple.java]: OneToOne association between persons was my first idea to design this association between persons (I didn't remember yet the Bytecode enhancement workaround with @LazyToOne(LazyToOneOption.NO_PROXY) ;-( )
With your explanation, I remember this crappy workaround and the problem of additional queries. So, I understand why you have designed like this. But a question remains: what do you think about @MapsId which [avoid  the bidirectional association](https://vladmihalcea.com/2016/07/26/the-best-way-to-map-a-onetoone-relationship-with-jpa-and-hibernate/) and byte-code enhancements?

Comment (jnizet) [reply backend/src/main/java/org/globe42/domain/Couple.java]: It only avoid byte-code enhancement if the association is not optional. If it's optional, Hibernate still has no way to know if a post has a post detail, and it must still check if a post detail with the same ID as the post exists or not.

Besides, in this case, we can't possibly associate two persons in a couple by making them have the same ID.

**Comment by jnizet** [comment]
Good catch. I knew I was forgetting something. This should be fixed now. I agree an alert would be nice, but I will leave that for later.

File: backend/src/main/java/org/globe42/domain/Couple.java (hunk 1 above)
Comment (acrepet) [reply backend/src/main/java/org/globe42/domain/Couple.java]: OK I got it

**Comment by acrepet** [comment]
OK perfect, everything is OK now. I've just filled an enhancement issue #141  for the alert and for an other thing about the correlation and the spouse and the MaritalStatus (for later)

**Pull request merged**